src/Containers/Microsoft.NET.Build.Containers/ContainerBuilder.cs

```diff
@@ -113,10 +113,15 @@ public static class ContainerBuilder
             // ports are validated by System.CommandLine API
             imageBuilder.ExposePort(number, type);
         }
+
         if (containerUser is { Length: > 0 } user)
         {
             imageBuilder.SetUser(user);
         }
+
+        ContainerHelpers.AssignUserFromEnvironment(imageBuilder, logger);
+        ContainerHelpers.AssignPortsFromEnvironment(imageBuilder, logger);
```

Comment:
as discussed on call, consider moving those to `ImageBuilder` to avoid code (and therefore test duplication) in `ContainerBuilder` and `CreateNewImage`.